### ISSUE TITLE
Not creating any object ID when posting

### ISSUE THREAD

**Issue by JHereu** [balderdashy/sails-postgresql#154]
Hey there,

I have a PostgreSQL database installed on localhost and a very simple Sails 0.11 app that wants to access the info storaged via sails-postgresql adapter.

I guess the database is set correctly, cause when I post a new object to the Sails app, it stores the data correcty but apparently without any id field. I've already defined a varchar column in my db table named 'id', as two more named 'createdAt' and 'updatedAt' (these two working correctly).

My model field 'autoPK' is not false and the Sails console logs the following error:

    error: Invalid usage of publishCreate() :: Values must have an `id`, instead got ::

Hope you could give me some tips.

Thanks :)

**Comment by tjwebb** [comment]
Post your actual model definition

**Comment by JHereu** [comment]
It's a very simple model definition, just a model 'Film.js' with 2 string attributes: 'title' and 'director', just to run the test.

    module.exports = {
        attributes: {
            'title': 'STRING',
            'director': 'STRING'
        }
    };

On the other hand, my PostgreSQL table has the following fields, in order to fill them via Sails:

      Column   |         Type          | Modifiers 
    -----------+-----------------------+-----------
     title     | character varying(30) | 
     director  | character varying(30) | 
     createdAt | character varying(30) | 
     updatedAt | character varying(30) | 
     id        | character varying(30) | 

Shall I mark the 'id' field on the table as primary key?

Thanks.

**Comment by segarcialo** [comment]
I have the same problem. How can I fix it? any idea?

I'm using NodeJS 0.10.35 64 bit and Sails 0.12 on Windows 7 machine with last stable version of PostgreSQL.

Thank you in advance!

**Comment by JHereu** [comment]
I guess I finally managed to solve the problem. Apparently, sails-postgre does not create a primary key for you as sails-mongo does (creating an _id column). So I just created an autoincremental column in the postgre table named *id* executing the following command:

    alter table myTable add column id bigserial primary key;

Bigserial is a "not null" bigInt that sets an autoincremental number by default once an entry is stored. So while postgre is managing the id,it doesn't mather if you set the _autoPK_ value of the model in Sails as true or false...

So problem solved. Hope this works for you too, @segarcialo.
